props/saber_fett263_buttons.h

```diff
@@ -1032,6 +1022,325 @@ class GestureControlFile : public ConfigFile {
+    saved_gesture_control.WriteToRootDir("gesture");
+  }
+  void say() { sound_library_.SayForcePush(); }
+};
+
+template<class SPEC>
+struct ForcePushLength : public SPEC::SteppedMode {
+  void mode_activate(bool onreturn) override {
+    push_ = saved_gesture_control.forcepushlen;
+    SPEC::SteppedMode::mode_activate(onreturn);
+    say();
+  }
+  void next() override {
+    push_ += 1;
+    if (push_ >= 10) push_ = 10;
+  }
+  void prev() override {
+    push_ -= 1;
+    if (push_ <= 1) push_ = 1;
+  }
+  void fadeout(float len) override {
+    mode::getSL<SPEC>()->fadeout(len);
+  }
+  void say() {
+    if (push_ == 1) sound_library_.SayMinimum();
+    if (push_ == 10) sound_library_.SayMaximum();
+    sound_library_.SayNumber(push_, SAY_WHOLE);
+  }
+  void select() { 
+    sound_library_.SayForcePushLength();
+    sound_library_.SayNumber(push_, SAY_WHOLE);
+    saved_gesture_control.forcepushlen = push_;
+    saved_gesture_control.WriteToRootDir("gesture");
+    SPEC::SteppedMode::select();
+  }
+  void exit() override {
+    mode::getSL<SPEC>()->SayCancel();
+    SPEC::SteppedMode::exit();
+  }
+
+int push_;
+};
+
+template<class SPEC>
+struct TwistOffGesture : public mode::BoolSetting {
+  bool get() { return saved_gesture_control.twistoff;  }
+  void set(bool value) { 
+    saved_gesture_control.twistoff = value;
+    saved_gesture_control.WriteToRootDir("gesture");
+  }
+  void say() { sound_library_.SayTwistRetraction(); }
+};
+
+template<class SPEC>
+struct PowerLock : public mode::BoolSetting {
+  bool get() { return saved_gesture_control.powerlock;  }
+  void set(bool value) { 
+    saved_gesture_control.powerlock = value;
+    saved_gesture_control.WriteToRootDir("gesture");
+  }
+  void say() { sound_library_.SayPowerLock(); }
+};
+
+template<class SPEC>
+struct LockupDelay : public SPEC::SteppedMode {
+  void mode_activate(bool onreturn) override {
+    delay_ = saved_gesture_control.lockupdelay;
+    SPEC::SteppedMode::mode_activate(onreturn);
+    say();
+  }
+  void next() override {
+    delay_ += 100;
+    if (delay_ >= 1200) delay_ = 1200;
+  }
+  void prev() override {
+    delay_ -= 100;
+    if (delay_ <= 200) delay_ = 200;
+  }
+  void fadeout(float len) override {
+    mode::getSL<SPEC>()->fadeout(len);
+  }
+  void say() {
+    if (delay_ == 1) sound_library_.SayMinimum();
+    if (delay_ == 10) sound_library_.SayMaximum();
+    sound_library_.SayNumber(delay_, SAY_WHOLE);
+    sound_library_.SayMillis();
+  }
+  void select() { 
+    sound_library_.SayLockupDelay();
+    sound_library_.SayNumber(delay_, SAY_WHOLE);
+    sound_library_.SayMillis();
+    saved_gesture_control.lockupdelay = delay_;
+    saved_gesture_control.WriteToRootDir("gesture");
+    SPEC::SteppedMode::select();
+  }
+  void exit() override {
+    mode::getSL<SPEC>()->SayCancel();
+    SPEC::SteppedMode::exit();
+  }
+    
+int delay_;
+};
+
+template<class SPEC>
+struct ClashDetect : public SPEC::SteppedMode {
```

Comment:
This shares a lot of code with ForcePushLength (since they both have 10 steps)
Maybe make a base class template?

Reply:
I'm not sure how to do, if you can provide the code I can try

Reply:
maybe we figure that out later